wgpu-core/src/command/mod.rs

```diff
@@ -214,10 +215,7 @@ impl CommandEncoderStatus {
                 Err(EncoderStateError::Ended)
             }
             Self::Locked(_) => Err(self.invalidate(EncoderStateError::Locked)),
-            st @ Self::Error(CommandEncoderError::State(EncoderStateError::Ended)) => {
-                *self = st;
-                Err(EncoderStateError::Ended)
-            }
```

Comment:
Here and below in `unlock_and_record` we still need to set the state back or else it will remain as `Transitioning`. I will fixup the commit.